tests/test_epos.py

```diff
@@ -0,0 +1,90 @@
+from impy.kinematics import CenterOfMass
+from impy.models import EposLHC
+from impy.constants import GeV
+import numpy as np
+from numpy.testing import assert_allclose
+from .util import (
+    reference_charge,
+    run_in_separate_process,
+    skip_on_ci_if_model_is_incompatible,
+)
+import pytest
+from particle import literals as lp
+from functools import lru_cache
+
+skip_on_ci_if_model_is_incompatible(EposLHC)
+
+
+def run_pp_collision():
+    evt_kin = CenterOfMass(10 * GeV, "proton", "proton")
+    m = EposLHC(evt_kin, seed=4)
+    m.set_stable(lp.pi_0.pdgid, True)
+    for event in m(1):
+        pass
+
+    # some methods only work on original event
+    assert event.impact_parameter > 0
+    assert event.n_wounded_A == 1
+    assert event.n_wounded_B == 1
+    assert event.n_wounded == 2
+
+    return event
+
+
+@pytest.fixture
+@lru_cache(maxsize=1)  # EposLHC initialization is very slow
+def event():
+    return run_in_separate_process(run_pp_collision)
+
+
+def test_charge(event):
```

Comment:
The test fails for me. I also don't quite understand what it does. Checks for charge conservation should be run on particles or particle like objects. Remnants and comment blocks should not be part of it.

Reply:
Edit: ...fails after fixing EPOS for nuclei since now there are nuclear remnants on the event stack with PDG codes that don't make the round trip through epos charge routines. A quick fix would be to only perform the check on charge conservation on final state particles. In all models it should be exact.

Reply:
Yes, that makes sense. The idea of this test for me was: if the particle is known to `particle`, ie. it is not a generator-specific pseudoparticle, then the charge that particle gives you should be the same as the charge that the model returns. If we go for what I am suggesting, namely that we compute the charge in an unified way from the pdg id for all models in the same way (make EventData.charge a property) then this test will automatically succeeed.